packages/template/src/interbit/private/actionTypes.js

```diff
@@ -0,0 +1,18 @@
+const covenantName = 'template-private'
+
+const actionTypes = {
+  MEMO: `${covenantName}/MEMO`,
+  ADD: `${covenantName}/ADD`
+}
+
+// Export a proxy so an exception is thrown in case an undefined property is accessed
+module.exports = new Proxy(actionTypes, {
+  get: (obj, prop) => {
+    if (prop in obj) {
+      return obj[prop]
+    }
+    throw new Error(
+      `Invalid action type "${prop}" in covenant "${covenantName}"`
+    )
+  }
+})
```

Comment:
This could get repetitive for a lot of covenants. However, I am all for using this as a 'best practice' pattern in the template.

Reply:
There are a few cases where this pattern could be applied.  For example, the actions themselves.

Although I agree with you:  In the future, I'd prefer to use named exports.  It's simpler, allows earlier detection of problems, and completely removes boilerplate.